Modules/_io/_iomodule.c

```diff
@@ -479,7 +480,13 @@ _io_text_encoding_impl(PyObject *module, PyObject *encoding, int stacklevel)
                 return NULL;
             }
         }
-        return &_Py_ID(locale);
```

Comment:
@ericsnowcurrently We need to incref this until we implement immortal object. Am I correct?

Reply:
Sort of.  Currently all statically allocated global objects have their refcount initialized to a really large value.  So we haven't been bothering with the incref if we know the object is one of those, which it is in this case.  Incref'ing it won't hurt.  However, it is effectively unnecessary.

Reply:
I am afraid breaking refleak buildbot.
Would you tell my why this `return &_Py_ID(locale);` don't break refleak test?

Reply:
The refcount of static objects is not directly included in `_Py_RefTotal`, which is used for refleak detection.  (`_Py_RefTotal` will still ensure that refcount operations are balanced though, regardless of the objects involved.)

Reply:
How is it achieved?

https://github.com/python/cpython/blob/cd05d0a423d97be69f9de4650f68f89e99ad68d1/Include/object.h#L473-L515

Py_INCREF and Py_DECREF don't check that the object is statically allocated.

Reply:
> (_Py_RefTotal will still ensure that refcount operations are balanced though, regardless of the objects involved.)

So skipping INCREF here might cause in-balance of `_Py_RefTotal`. Doesn't it break refleak test?